### PR TITLE
Swap map service

### PR DESCRIPTION
**Pull request by liz-murphy**
Adds a service which a client can use to send a new map (nav_msgs/OccupancyGrid) and new initial pose (geometry_msgs/Pose) for amcl to load and localize against.

### CODE REVIEW

File: amcl/package.xml [hunk 1]
```diff
@@ -29,10 +29,16 @@
...
     <build_depend>tf</build_depend>
+    <build_depend>nav_msgs</build_depend>
+    <build_depend>geometry_msgs</build_depend>
+    <build_depend>message_generation</build_depend>
 
     <run_depend>roscpp</run_depend>
     <run_depend>dynamic_reconfigure</run_depend>
     <run_depend>tf</run_depend>
+    <run_depend>nav_msgs</run_depend>
+    <run_depend>run_msgs</run_depend>
```
Comment (mikeferguson) [inline amcl/package.xml]: I presume this is supposed to be geometry_msgs

File: amcl/srv/SwapMap.srv [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ nav_msgs/OccupancyGrid map_msg
```
Comment (mikeferguson) [inline amcl/srv/SwapMap.srv]: I suggest this be called "map" rather than "map_msg" -- we already know it is a message.

**Comment by mikeferguson** [comment]
A couple of general comments:
- I would suggest not putting a new service message definition here in AMCL -- I actually think that should be upstreamed into nav_msgs as SetMap (with the same definition you have right now) as there is already a GetMap service (http://docs.ros.org/api/nav_msgs/html/srv/GetMap.html).
- If we decide to keep the message here, I see no dependency on the generated messages in your CMake -- this means it may or may not build properly on the farm.

**Comment by liz-murphy** [comment]
Fixed issues and created a corresponding PR for the service message in nav_msgs: https://github.com/ros/common_msgs/pull/51

**Comment by mikeferguson** [comment]
This is pending upstream merge of common_msgs change -- and will also need to be retargeted to the proper branch (let's wait and see if we can get this into indigo-devel -- I'm all for it on nav side, we just have to see about the message backport).

**Comment by stwirth** [comment]
This has to be adapted to use a PoseWithCovarianceStamped.

**Comment by stwirth** [comment]
I hacked on it, PR following now.

**Comment by stwirth** [comment]
Superseded by #321.

**Comment by mikeferguson** [comment]
It looks like the messages never got into hydro-devel? @stwirth are you planning on getting that merged (and released) before hydro gets EOL, or should we close this PR?

**Comment by stwirth** [comment]
@mikeferguson close please!

**Pull request closed without merging**